components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaProducer.java

```diff
@@ -194,11 +202,11 @@ public ProducerRecord next() {
                     Object value = tryConvertToSerializedType(exchange, next, endpoint.getConfiguration().getSerializerClass());
 
                     if (hasPartitionKey && hasMessageKey) {
-                        return new ProducerRecord(msgTopic, partitionKey, key, value);
+                        return new ProducerRecord(msgTopic, partitionKey, null, key, value, propagatedHeaders);
```

Comment:
can you replace null with named constants?

Reply:
a lot of code would be added only for this purpose, is there a need for this?

Reply:
yes, this is better than new MyClass(null, null, null, null)...